onnx/utils.py

```diff
@@ -234,12 +235,20 @@ def extract_model(
     model = onnx.load(input_path)
 
     if infer_shapes:
-        model = onnx.shape_inference.infer_shapes(model)
+        if model.ByteSize() > MAXIMUM_PROTOBUF:
+            onnx.shape_inference.infer_shapes_path(input_path, output_path)
+            model = onnx.load(output_path)
+        else:
+            model = onnx.shape_inference.infer_shapes(model)
 
     e = Extractor(model)
     extracted = e.extract_model(input_names, output_names)
 
-    onnx.save(extracted, output_path)
+    if model.ByteSize() > MAXIMUM_PROTOBUF:
```

Comment:
A model could be below MAXIMUM_PROTOBUF and above after shape inference. I would at least document this new behaviour wherever this makes sense.

Reply:
Added note
```
Note: When the extracted model size is larger than 2GB, the extra data will be saved in "output_path.data".
```